### ISSUE TITLE
[Question] RAC3 / Swift2: UI binding

### ISSUE THREAD

**Issue by harrywincup** [ReactiveCocoa/ReactiveCocoa#2282]
Hi. I'm _very_ new to the ReactiveCocoa scene so apologies if these are silly questions!

I'm attempting to use the Swift API (as suggested in the Swift2 readme), but i'm confused about whether i'm doing my UI binding in my `UITableViewCell` subclass correctly. I have a view model optional (which i've seen suggested in other issues) 
```
let viewModel: MutableProperty<FeedViewModel?> = MutableProperty(nil)
```
To bind a value to a `UILabel`, is there a concise way to use `<~` still, or is the Swift API way™ to start the vm producer and manually set the label text like so (which loses some of the declarativeness?):

```
viewModel.producer
	.map { $0?.name.value }
	.start(next: {
		self.feedNameLabel.text = $0
	})
```

The same question for `Action`s ... I have a `UISwitch` inside the table cell. The `Action` is set up in the view model but the way i'm binding it in the view feels very wrong (saw this approach in another issue):

```
viewModel.producer.flatMap(.Latest) {
	return SignalProducer(value: $0?.subscribeAction.unsafeCocoaAction ?? nil)
}.ignoreNil().start(next: {
	self.subscriptionSwitch.addTarget($0, action: CocoaAction.selector, forControlEvents: .ValueChanged)
})
```

That approach requires that I manually `removeTarget` from the switch inside the cell's `prepareForReuse` ... I've seen in most example code people using `cell.rac_prepareForReuse` signal, but that's the old API right so i'm not sure on how to get/use a reuse signal to avoid manually removing the action?

**Any** pointers for anything i've mentioned would be much appreciated! Thanks :)

**Comment by harrywincup** [comment]
Or maybe i'm supposed to use `DynamicProperty` instead? Although everywhere says that's a very last resort. Is binding to UIControls that don't have RAC extensions a "last resort"?

It seems to work at least ...

```
DynamicProperty(object: feedNameLabel, keyPath: "text") <~ viewModel.producer.ignoreNil().map { $0.name.value }
```

**Comment by andersio** [comment]
IMO View subclasses should not be aware of things other than its own presentation layout.

So I would instead expose on the View subclasses binding points as `MutableProperty`s or custom extensions on `UIControl`s, or use the existing ObjC RAC signals. Then you can bind cold signals of properties of your ViewModel to these binding points in the ViewController, which is supposed to mediate between the ViewModel and Views. For a UITableView, `tableView(_:cellForRowAtIndexPath:)` is a good place to perform your `<~` bindings.

In this way, you also decouple the view from a particular ViewModel.

**Comment by harrywincup** [comment]
@andersio Thanks for the help, much appreciated :)

**Comment by andersio** [comment]
Recently I hit kind of a wall when I tried to clean up the memory management (tried some dirty way and failed, ugh), and I would like to reverse course on my "suggestion". Letting view subclasses own its detail view model sounds a better approach to me now, since you can associate the life cycle of it to the view itself in a cleaner way.

But for UITableView, I still think there is no need to bind view model to a cell via MutableProperty. You can create them on the fly from the parent view model, and inject them into the cell. If you have insertion or deletion in the data collection, you can use the row manipulation method to keep the table view consistent too (with animations!).

**Comment by andersio** [comment]
Sorry, I stand corrected - UIView subclasses are indeed a more ideal place to retain its associated view model. I found my suggestion, per what I did, leads to a sort of dirty solution (managing a collection of view models in parallel) in my code base, so after a few prototyping I decided to refactor it.

You may make use of 'rac_prepareForReuseSignal' to bind your ViewModel and properties by the way.
